rfc/trace_identifiers.md

```diff
@@ -0,0 +1,86 @@
+
+## Trace Observers
+The OpenTracing community would like to develop secondary observation systems which utilize the tracing runtime, but are tracer independent. Examples include:
+
+* Generating metrics from tracing data.
+* Integrating with runtime and system diagnostics.
+* Integrating with 3rd-party context-propagation systems.
+* Correlating logs, as mentioned above.
+
+# Risk Assessment
+Because this proposal includes the exposure of new information, and adds entirely new concepts to the interface, some risks exist.
+
+## Tracer support
+Some existing tracers may not be able to support this feature, as their internal model does not include any client-side trace identifiers. These tracers may choose to not support this feature by returning empty string values.
+
+## Protocol support
+It's possible that new tracing protocols may emerge which use an entirely different header scene. Examples could include a tracing system which handles trace joins explicitly as part of the protocol, and thus no longer has an equivalent concept of a trace id, or a system which uses backpropagation to contain additional data.
+
+This is mitigated by the fact that the concept of a span and a trace are directly part of the OpenTracing model of computation. Some form of identifier for these objects will  be availiable to a tracer that conforms to this model of computation. While its likely that additional identifiers or interfaces may be necessary to handle future changes, it is impossible that the trace and span concepts will be removed from this version of OpenTracing. 
+
+Because the accessors produce a variable-width string value, new formats and wireprotocols for these identifiers will not result in a breaking change for the OpenTracing interface. Likewise, systems which consume this data are by definition separate from the tracing system, and are not dependent on the format of the identifier.
+
+## Extra Allocations and Overhead
+Internally, tracers do not always use strings to represent their identifiers. So there is a conversion cost when using these accessors. 
+
+While a single allocation may be inevitable, exposing accessors in additional formats could be done to prevent double allocations while formating the identifiers. For example, converting from a tracer’s native format to a string may trigger an allocation. If there are many systems which want to consume the identifier in a format which requires an allocation when converting from a string, a second allocation could occur.
```

Comment:
What I picture for a C API would not incur allocation in the library. Just ask the user for a buffer.

```
#define TRACE_ID_STR_LEN 16
int span_context_trace_id(const custom_span_context* ctx, void* buf, int len)
{
    return snprintf(buf, len, "%" PRIx64 "%016" PRIx64, span_context->trace_id.high, span_context->trace_id.low);
}
```

Reply:
Good approach! In many languages, this option will probably not be available, as the APIs you are trying to put the data into often will not support buffer types (or the language may not have them at all), but for C this is good.